packages/framework/src/Model/Article/ArticleDetailFriendlyUrlDataProvider.php

```diff
@@ -54,9 +54,20 @@ public function getFriendlyUrlData(DomainConfig $domainConfig): array
         $friendlyUrlsData = [];
 
         foreach ($scalarData as $data) {
-            $friendlyUrlData = $this->friendlyUrlDataFactory->create();
-            $friendlyUrlData->name = $data['id'];
-            $friendlyUrlData->id = $data['name'];
+            if (method_exists($this->friendlyUrlDataFactory, 'createFromIdAndName')) {
+                $friendlyUrlData = $this->friendlyUrlDataFactory->createFromIdAndName($data['id'], $data['name']);
+            } else {
+                @trigger_error(
+                    sprintf(
+                        'This approach is deprecated since Shopsys Framework 7.3, implement and use "%s" instead',
+                        FriendlyUrlDataFactoryInterface::class . '::createFromIdAndName()'
+                    ),
+                    E_USER_DEPRECATED
+                );
+                $friendlyUrlData = $this->friendlyUrlDataFactory->create();
+                $friendlyUrlData->id = $data['id'];
+                $friendlyUrlData->name = $data['name'];
+            }
```

Comment:
The whole method is quite hard to read. Can you please extract some parts into methods?
I thought about 
```php 
if (method_exists($this->friendlyUrlDataFactory, 'createFromIdAndName')) {
    return $this->getDataTheNewWay();
} else {
    return $this->getDataTheOldFashionWay();
}
```
I know there will be for loop in both methods, but It helps extensibility and readability of a code.
What do you think?

_Same applies for all the Data providers_

Reply:
fair point, fixed